lib/vmpooler/providers/vsphere.rb

```diff
@@ -579,6 +585,65 @@ def memory_utilization_for(host)
           (memory_usage.to_f / memory_size.to_f) * 100
         end
 
+        def get_average_cluster_utilization(hosts)
+          utilization_counts = hosts.map { |host| host[0] }
+          utilization_counts.inject(:+) / hosts.count
+        end
+
+        def build_compatible_hosts_lists(hosts, percentage)
+          hosts_with_arch_versions = hosts.map { |host| [host[0], host[1], get_host_cpu_arch_version(host[1])] }
+          versions = hosts_with_arch_versions.map { |host| host[2] }.uniq
+          architectures = {}
+          versions.each do |version|
+            architectures[version] = []
+          end
+
+          hosts_with_arch_versions.each do |host|
+            architectures[host[2]] << [host[0], host[1], host[2]]
```

Comment:
I find it hard to know what is in each index `host[0], host[1], host[2]`

Reply:
Yeah, it probably makes sense to put it in a hash and give these values some titles.

Reply:
This now uses a hash which should make clear what's being placed into this array.